src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1435,10 +1441,10 @@ private MemberDeclarationSyntax ParseTypeDeclaration(SyntaxList<AttributeListSyn
 
 #nullable enable
 
-        private TypeDeclarationSyntax ParseClassOrStructOrInterfaceDeclaration(SyntaxList<AttributeListSyntax> attributes, SyntaxListBuilder modifiers)
+        private TypeDeclarationSyntax ParseClassOrStructOrInterfaceOrRoleDeclaration(SyntaxList<AttributeListSyntax> attributes, SyntaxListBuilder modifiers)
```

Comment:
>ParseClassOrStructOrInterfaceOrRoleDeclaration [](http://example.com/codeflow?start=38&length=46)

`...OrRecordOrRoleOrExtensionDeclaration`?

Reply:
Maybe we're reaching a point where this name is no longer scaling :P

Reply:
Records are either a class or a struct. Extensions are roles. That's the reason for current list.